calplus/v1/compute/drivers/openstack.py

```diff
@@ -0,0 +1,171 @@
+""" OpenstackDriver for Compute
+    based on BaseDriver
+"""
+
+
+from datetime import datetime
+import six
+
+from keystoneauth1.identity import v3
+from keystoneauth1 import session
+from novaclient.client import Client
+
+from calplus.v1.compute.drivers.base import BaseDriver, BaseQuota
+
+
+PROVIDER = "OPENSTACK"
+
+
+class OpenstackDriver(BaseDriver):
+    """docstring for OpenstackDriver"""
+
+    def __init__(self, cloud_config):
+        super(OpenstackDriver, self).__init__()
+        self.auth_url = cloud_config['os_auth_url']
+        self.project_name = cloud_config['os_project_name']
+        self.username = cloud_config['os_username']
+        self.password = cloud_config['os_password']
+        self.user_domain_name = \
+            cloud_config.get('os_project_domain_name', 'default')
+        self.project_domain_name = \
+            cloud_config.get('os_user_domain_name', 'default')
+        self.driver_name = \
+            cloud_config.get('driver_name', 'default')
+        self.tenant_id = cloud_config.get('tenant_id', None)
+        self.limit = cloud_config.get('limit', None)
+        self.client_version = \
+            cloud_config.get('os_novaclient_version', '2.1')
```

Comment:
@daikk115 Đại THE FALLEN, where the hell did you define all these config options?